omniduct/databases/base.py

```diff
@@ -35,10 +35,18 @@ def render_statement(method, self, statement, *args, **kwargs):
         pass
 
     # If templating enabled, render template
-    template = kwargs.pop('template', False)
-    template_context = kwargs.pop('template_context', {})
-
-    if template:
+    if kwargs.pop('template', False):
+        template_context = {}
+        template_context.update(self._template_context)
+        kwarg_context = kwargs.pop('template_context', {})
+        template_context.update(kwarg_context)
+        intersection = set(self._template_context.keys()) & set(kwarg_context.keys())
+        if intersection:
+            logger.warning(
+                'Duct default context conflicts with the context passed in '
```

Comment:
I think this logging message has the potential to confuse some people, as templates are never mentioned.

How about: "The following default template context keys have been overridden by the local context: {}."